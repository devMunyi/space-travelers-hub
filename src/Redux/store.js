import { configureStore, applyMiddleware } from '@reduxjs/toolkit';
import logger from 'redux-logger';
import rocketsReducers from './reducers/rockets';

export default configureStore(
  {
    reducer: {
      rockets: rocketsReducers,
    },
  },
  applyMiddleware(logger)
);
