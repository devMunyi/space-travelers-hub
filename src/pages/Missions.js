import React, { useEffect } from "react";
import Table from "react-bootstrap/Table";
import { useDispatch, useSelector } from "react-redux";
import {
  fetchMissions,
  bookedMission,
  cancelledMission,
} from "../redux/mission/missions";

const Missions = () => {
  const dispatch = useDispatch();

  useEffect(() => {
    dispatch(fetchMissions());
  }, [dispatch]);

  const missions = useSelector((state) => state.missions);

  const handleBookMission = (id) => {
    dispatch(bookedMission(id));
  };

  const handleCancelMission = (id) => {
    dispatch(cancelledMission(id));
  };

  return (
    <Table striped bordered hover>
      <thead>
        <tr>
          <th>Description</th>
          <th>Status</th>
          <th>Action</th>
        </tr>
      </thead>
      <tbody>
        {missions.map((mission) => (
          <tr key={mission.mission_id}>
            <td>{mission.mission_name}</td>
            <td>{mission.description.slice(0, 200)}...</td>
            <td>
              <p
                style={{
                  background: mission.reserved ? "blue" : "gray",
                  color: mission.reserved ? "white" : "white",
                }}
                className='active rounded p-2'
              >
                {mission.reserved ? "Active Member" : "NOT A MEMBER"}
              </p>
            </td>
            <td>
              {mission.reserved && (
                <button
                  style={{ border: "red 1px solid", color: "red" }}
                  onClick={() => handleCancelMission(mission.mission_id)}
                  type='button'
                  className='btn btn-sm btn-outline-secondary rounded p-2'
                >
                  Leave mission
                </button>
              )}
              {!mission.reserved && (
                <button
                  onClick={() => handleBookMission(mission.mission_id)}
                  type='button'
                  className='btn btn-sm btn-primary rounded p-2'
                >
                  Join mission
                </button>
              )}
            </td>
          </tr>
        ))}
      </tbody>
    </Table>
  );
};

export default Missions;
