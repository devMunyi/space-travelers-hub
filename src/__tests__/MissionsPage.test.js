import React from 'react';
// import '@testing-library/jest-dom';
import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import store from '../redux/store';
import Missions from '../pages/Missions';

it('renders Component correctly', () => {
  const tree = renderer
    .create(
      <Provider store={store}>
        <Missions />
      </Provider>,
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
