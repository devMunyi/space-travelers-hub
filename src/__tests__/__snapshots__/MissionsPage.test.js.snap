// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders Component correctly 1`] = `
<table
  className="container table table-striped table-bordered table-hover"
>
  <thead>
    <tr>
      <th>
        Description
      </th>
      <th>
        Status
      </th>
      <th>
        Action
      </th>
    </tr>
  </thead>
  <tbody />
</table>
`;
